### PR TITLE
Split `MeasureUnit` into `MeasureUnitParser` and `MeasureUnit`

### PR DESCRIPTION
**Pull request by younies**

### CODE REVIEW

**Review by sffc — DISMISSED** [review]
Good for experimental. Looking forward to when we start running it with data

File: experimental/unitsconversion/src/measureunit.rs
Comment (robertbastian) [inline experimental/unitsconversion/src/measureunit.rs:24]: issue: the public API should construct this from the data struct, through a provider. this looks like a datagen-only API, so it should be behind the datagen feature

File: experimental/unitsconversion/src/measureunit.rs
Comment (robertbastian) [inline experimental/unitsconversion/src/measureunit.rs:19]: nit: don't repeat type information in names

File: experimental/unitsconversion/src/measureunit.rs
Comment (robertbastian) [inline experimental/unitsconversion/src/measureunit.rs:136]: this is done?

File: experimental/unitsconversion/src/measureunit.rs
Comment (robertbastian) [inline experimental/unitsconversion/src/measureunit.rs:1]: observation: there are no unit tests for this code?

**Comment by robertbastian** [comment]
Consider renaming the PR to "Split MeasureUnit into MeasureUnitParser and MeasureUnit". You're not really implementing anything here.

File: experimental/unitsconversion/src/measureunit.rs
Comment (younies) [reply experimental/unitsconversion/src/measureunit.rs:20]: done

File: experimental/unitsconversion/src/measureunit.rs
Comment (younies) [reply experimental/unitsconversion/src/measureunit.rs:1]: I added a `todo` to test it after adding the UnitsTest.txt to the test data.

Comment (younies) [reply experimental/unitsconversion/src/measureunit.rs:1]: It is in the upcoming PR.

File: experimental/unitsconversion/src/measureunit.rs
Comment (younies) [reply experimental/unitsconversion/src/measureunit.rs:25]: let's discuss this API before submission in the ICU4X meeting today.

File: experimental/unitsconversion/src/measureunit.rs
Comment (younies) [reply experimental/unitsconversion/src/measureunit.rs]: yes :)

File: experimental/unitsconversion/src/measureunit.rs
Comment (robertbastian) [reply experimental/unitsconversion/src/measureunit.rs:1]: I'd consider those end to end integration tests. You should be able to write some simple tests that verify that this code works before it's merged.

File: experimental/unitsconversion/src/measureunit.rs
Comment (Manishearth) [reply experimental/unitsconversion/src/measureunit.rs:25]: On talking to Younies: the way to construct the Parser should be a method from the MeasureUnitConversionFactory which is what owns the data

We should:

 - Temporarily mark this `cfg(feature = datagen)` (since it's still needed by datagen)
 - Rename it to `from_payload()`
 - Add a todo to revisit the public nature of the API
 - (separate PR) make this constructable from a Factory

File: experimental/unitsconversion/src/measureunit.rs
Comment (robertbastian) [reply experimental/unitsconversion/src/measureunit.rs:1]: Actually I got mixed up with the other PR, you're not adding any code here. While it's not great that this code is not tested, it shouldn't block this PR

File: experimental/unitsconversion/src/measureunit.rs
Comment (robertbastian) [reply experimental/unitsconversion/src/measureunit.rs:20]: also in the field name

Comment (younies) [reply experimental/unitsconversion/src/measureunit.rs]: done

File: experimental/unitsconversion/src/measureunit.rs
Comment (younies) [reply experimental/unitsconversion/src/measureunit.rs:1]: Okay, np

File: experimental/unitsconversion/src/measureunit.rs
Comment (younies) [reply experimental/unitsconversion/src/measureunit.rs]: done

**Pull request merged**
